### PR TITLE
Add properties of the Bayes risk of a prior

### PR DESCRIPTION
**Pull request by LorenzoLuccioli**
- Add the file `KernelConstComp.lean` with lemmas about the composition of two kernels when the first one is constant.
- Add definition of `estimationProblem.compProd`.
- Add properties of the Bayes risk of a prior: `bayesRiskPrior_compProd_le_bayesRiskPrior`, `bayesRiskPrior_le_inf`, `bayesianRisk_eq_bayesInv_prod`, `bayesianRisk_ge_lintegral_iInf_bayesInv`.
- Add the DPI for the Bayes risk increase.

### CODE REVIEW

File: TestingLowerBounds/Testing/Risk.lean [hunk 1]
```diff
@@ -153,6 +167,65 @@ lemma bayesRisk_le_minimaxRisk (E : estimationProblem Θ 𝒳 𝒴 𝒵) :
...
+    congr with θ
+    rw [lintegral_dirac']
+    have := E.ℓ_meas
+    fun_prop [E.ℓ_meas]
+  · exact kernel.isMarkovKernel_const
+
+lemma bayesianRisk_eq_bayesInv_prod [StandardBorelSpace Θ] [Nonempty Θ]
+    (E : estimationProblem Θ 𝒳 𝒴 𝒵) [IsMarkovKernel E.P] (κ : kernel 𝒳 𝒵)
+    (π : Measure Θ) [IsFiniteMeasure π] [IsSFiniteKernel κ] :
+    bayesianRisk E κ π = ∫⁻ (θz : Θ × 𝒵), E.ℓ (E.y θz.1, θz.2) ∂(π ∘ₘ (((E.P†π) ×ₖ κ) ∘ₖ E.P)) := by
```
Comment (RemyDegenne) [inline TestingLowerBounds/Testing/Risk.lean]: Can you add a docstring to that lemma to explain what it does? The result is a bit hard to parse.
Also the name could be instead `bayesianRisk_eq_lintegral_bayesInv_prod`.
Finally, it might be useful to have a version with that variant of the result (in which the measure in the integral is written slightly differently):
`bayesianRisk E κ π = ∫⁻ (θz : Θ × 𝒵), E.ℓ (E.y θz.1, θz.2) ∂((π ∘ₘ E.P) ∘ₘ ((E.P†π) ×ₖ κ))`
It will probably be useful because `π ∘ₘ E.P` is the distribution of the data.

Comment (LorenzoLuccioli) [reply TestingLowerBounds/Testing/Risk.lean]: I think we can just have the version you are proposing, without the other one with `(π ∘ₘ (((E.P†π) ×ₖ κ) ∘ₖ E.P))`, since it is only used once, and anyway if we need the previous version later we can easily switch between the two using `Measure.comp_assoc`.

**Review by RemyDegenne — APPROVED** [review]
